# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1203283668 -3600
#      Sun Feb 17 22:27:48 2008 +0100
# Node ID 61aee739a4da03f4064cd947b3b5812332467b77
# Parent  8ca8e97e8c0aef9132d581b74c67634ef2a2742c
Make sure to initialize axes xform data.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -10,16 +10,19 @@ 2008-06-04  Shai Ayal  <shaiay@users.sou
 
 	* graphics.h.in (axes::properties::update_xlim): call the new
 	axes::properties::calc_ticks_and_lims method
 	(axes::properties::update_ylim): ditto
 	(axes::properties::update_zlim): ditto
 
 2008-06-04  Michael Goffioul <michael.goffioul@gmail.com>
 
+	* graphics.h.in (axes::axes): Be sure to initialize transformation
+	data correctly.
+
 	* graphics.cc (xget_ancestor): Pass graphics_object argument by value
 	instead of by reference.
 
 	* graphics.h.in (surface::properties::xdata,
 	surface::properties::ydata, surface::properties::zdata,
 	surface::properties::normalmode, surface::properties::vertexnormals):
 	Add 'u' modifier.
 	(surface::properties::update_normals): New method to compute normals
diff --git a/src/graphics.h.in b/src/graphics.h.in
--- a/src/graphics.h.in
+++ b/src/graphics.h.in
@@ -2611,16 +2611,17 @@ public:
 private:
   properties xproperties;
 
 public:
   axes (const graphics_handle& mh, const graphics_handle& p)
     : base_graphics_object (), xproperties (mh, p), default_properties ()
   {
     xproperties.override_defaults (*this);
+    xproperties.update_transform ();
   }
 
   ~axes (void) { xproperties.delete_children (); }
 
   void override_defaults (base_graphics_object& obj)
   {
     // Allow parent (figure) to override first (properties knows how
     // to find the parent object).
